utils/oci.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2023 The kubecfg authors
+//
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package utils
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/StalkR/httpcache"
+	"github.com/containerd/containerd/remotes"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/pkg/auth/docker"
+)
+
+const (
+	OCIBundleBodyMediaType   = "application/vnd.kubecfg.bundle.tar+gzip"
```

Comment:
Will it always be gzipped? I saw your comment about `kubecfg --alpha pack` so I guess it won't matter.

Reply:
yes, oras builds a targz and the current code parses the content of the blob using gzip.Reader and tar.Reader